cli/src/pcluster/validators/secret_validators.py

```diff
@@ -0,0 +1,68 @@
+# Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+# with the License. A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+# OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions and
+# limitations under the License.
+import base64
+import binascii
+
+from pcluster.aws.aws_api import AWSApi
+from pcluster.aws.common import AWSClientError
+from pcluster.validators.common import FailureLevel, Validator
+
+
+class MungeKeySecretArnValidator(Validator):
+    """MungeKeySecretArn validator."""
+
+    def _validate(self, munge_key_secret_arn: str, region: str):
+        try:
+            arn_components = munge_key_secret_arn.split(":")
+            service = arn_components[2]
+            resource = arn_components[5]
+            if service == "secretsmanager" and resource == "secret":
```

Comment:
Some of the validators are similar to what we use for the [PasswordSecretArnValidator](https://github.com/aws/aws-parallelcluster/blob/1763c2a36f7bfe60f46acc81a02a86ff031eba78/cli/src/pcluster/validators/directory_service_validators.py#L86).

Is it possible to reuse the same validator?

Reply:
If you mean to reuse the common codes in a shared function, that's possible but seems not necessary. But to reuse the same validator is not possible I think. Some logics are not the same. This validator only need to check secretsmanager, the [PasswordSecretArnValidator](https://github.com/aws/aws-parallelcluster/blob/1763c2a36f7bfe60f46acc81a02a86ff031eba78/cli/src/pcluster/validators/directory_service_validators.py#L86) also check ssm. And this validator contains a logic of check the size of custom munge key and a logic of decoding custom munge key.

Reply:
Done! Now I am on refactoring unit tests for munge key related validators.